python/ray/scripts/scripts.py

```diff
@@ -463,9 +464,11 @@ def start(node_ip_address, address, port, redis_password, redis_shard_ports,
           system_config, enable_object_reconstruction, metrics_export_port,
           no_monitor, tracing_startup_hook, ray_debugger_external):
     """Start Ray processes manually on the local machine."""
-    if gcs_server_port and not head:
+    if use_gcs_for_bootstrap() and gcs_server_port is not None:
+        cli_logger.abort("`{}` is deprecated. Specify {} instead.",
+                         cf.bold("--gcs-server-port"), cf.bold("--port"))
         raise ValueError(
-            "gcs_server_port can be only assigned when you specify --head.")
+            "`--gcs-server-port` is deprecated. Specify `--port` instead.")
```

Comment:
Will this bring breaking change? Should we print a warning and use `--gcs-server-port` with higher priority?

Reply:
For users calling `ray start --gcs-server-port=7890`, with Redis bootstrapping Ray will start at 6379. If Ray uses 7890 for GCS port with GCS bootstrapping, and the user does not see the updated command to connect to Ray when they run `ray start`, their autoscaler configs or commands to start work nodes will break with connection failure. It seems better to always use `--port` for GCS port in GCS bootstrapping mode.